core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -778,11 +836,19 @@ public boolean isFollowing() {
         int lookaheadSize = -1;
         int sigsRequiredToSpend = 1;
 
+        List<ChildNumber> accountPath = newArrayList();
         PeekingIterator<Protos.Key> iter = Iterators.peekingIterator(keys.iterator());
         while (iter.hasNext()) {
             Protos.Key key = iter.next();
             final Protos.Key.Type t = key.getType();
             if (t == Protos.Key.Type.DETERMINISTIC_MNEMONIC) {
+                accountPath = newArrayList();
+                for (int i : key.getAccountPathList()) {
```

Comment:
Related to my other comment: I guess we need to guard for null here for if an old wallet is read?

Reply:
getAccountPathList() would just return an empty list in the case.

Reply:
Repeated Fields

For this field definition:

repeated int32 foo = 1;
The compiler will generate the following accessor methods in both the message class and its builder:

int getFooCount(): Returns the number of elements currently in the field.

int getFoo(int index): Returns the element at the given zero-based index.

List<Integer> getFooList(): 
Returns the entire field as a list. If the field is not set, returns an empty list. 
The returned list is immutable for message classes and unmodifiable for message builder classes.

https://developers.google.com/protocol-buffers/docs/reference/java-generated#repeated-fields